### PR TITLE
Bug report 2 issue template

### LINKED ISSUE #119918: Migrate the "Bug report" issue template to Github forms
We should migrate the existing new issue template for reporting a bug to use Github forms. This allows us to add validations for required fields allowing the quality raised issues to be improved.

### PR DESCRIPTION
**Pull request by exaby73**
This PR migrates the bug report template to Github Forms

Fixes https://github.com/flutter/flutter/issues/119918

<details>
<summary>Preview of new form</summary>

image

</details>

## Pre-launch Checklist

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [X] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [X] I signed the [CLA].
- [X] I listed at least one issue that this PR fixes in the description above.
- [X] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [X] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by darshankawar** [comment]
1. The template description can be improved as below:

`You found a bug in the application. It is crashing or throws an exception, a widget is buggy or something looks wrong.`

Screenshot 2023-03-13 at 6 24 00 PM

2. The first line seems repetitive, so we can remove it and just keep `If our documentation doesn't have an answer..` statement. 

3. `Code sample` field should be mandatory I think.

**Comment by darshankawar** [comment]
Screenshot 2023-03-14 at 6 47 36 PM


> 2\. The first line seems repetitive, so we can remove it and just keep `If our documentation doesn't have an answer..` statement.

You seem to have removed second statement and kept first, whereas, I suggested to do opposite. Because this line `If you have found a bug` is reptitive with description of the template, or we can simply remove this line altogether.

**Comment by exaby73** [comment]
Fixed it :)

**Comment by darshankawar** [comment]
LGTM.

File: .github/ISSUE_TEMPLATE/2_bug.yml
Comment (helinanever) [inline .github/ISSUE_TEMPLATE/2_bug.yml:45]: Missing full stop

File: .github/ISSUE_TEMPLATE/2_bug.yml
Comment (helinanever) [inline .github/ISSUE_TEMPLATE/2_bug.yml:55]: I think it should be "use the `flutter create bug` command"

File: .github/ISSUE_TEMPLATE/2_bug.yml
Comment (helinanever) [inline .github/ISSUE_TEMPLATE/2_bug.yml:61]: missing full stop

Comment (helinanever) [reply .github/ISSUE_TEMPLATE/2_bug.yml:61]: Actually, I would shorten the whole thing and list both alternatives in a single sentence, something like "Alternatively, you can use https://dartpad.dev/ or create a public GitHub repository to share your sample.”

File: .github/ISSUE_TEMPLATE/2_bug.yml
Comment (helinanever) [inline .github/ISSUE_TEMPLATE/2_bug.yml:102]: Github > GitHub

File: .github/ISSUE_TEMPLATE/2_bug.yml
Comment (stuartmorgan) [inline .github/ISSUE_TEMPLATE/2_bug.yml:71]: There's no need to use HTML comment format inside a code block, since it won't actually be treated as a comment.

File: .github/ISSUE_TEMPLATE/2_bug.yml
Comment (stuartmorgan) [inline .github/ISSUE_TEMPLATE/2_bug.yml:111]: Same here and belowe.

File: .github/ISSUE_TEMPLATE/2_bug.yml
Comment (stuartmorgan) [inline .github/ISSUE_TEMPLATE/2_bug.yml:20]: Is `description` required? If not, I would vote for removing this line since it seems redundant with the label and the first checkbox.

**Review by stuartmorgan-g — APPROVED** [review]
LGTM with some nits, I'm sorry for the long delay on this :(

File: .github/ISSUE_TEMPLATE/2_bug.yml
Comment (exaby73) [reply .github/ISSUE_TEMPLATE/2_bug.yml]: Done

File: .github/ISSUE_TEMPLATE/2_bug.yml
Comment (exaby73) [reply .github/ISSUE_TEMPLATE/2_bug.yml]: Changed all places

File: .github/ISSUE_TEMPLATE/2_bug.yml
Comment (exaby73) [reply .github/ISSUE_TEMPLATE/2_bug.yml]: Removed the description

**Comment by exaby73** [comment]
Fix the nits and rebased

**Comment by exaby73** [comment]
Rebased, now all checks pass. @Piinks maybe `autosubmit` can be added again?

**Comment by stuartmorgan** [comment]
FYI, you should be able to add the label yourself.

**Comment by exaby73** [comment]
Yes just not sure if that's allowed and if that's considered merging my own PR :) But thanks

**Comment by stuartmorgan** [comment]
Absolutely! Once you have approval on a PR, you're welcome to `autosubmit` it as soon as you're ready :)

**Pull request merged**
